app/src/main/java/org/mifos/mobile/utils/TransactionUiState.kt

```diff
@@ -0,0 +1,18 @@
+package org.mifos.mobile.utils
+
+import org.mifos.mobile.models.Transaction
+
+
+sealed class TransactionUiState {
+    data class Loading(val showProgress: Boolean) : TransactionUiState()
+    data class RecentTransactions(val transactions: List<Transaction>) : TransactionUiState()
+    data class LoadMoreTransactions(val transactions: List<Transaction>) : TransactionUiState()
+    object EmptyTransaction : TransactionUiState()
+    data class ErrorFetchingTransactions(val errorMessageResId: Int) : TransactionUiState()
+
+    companion object {
+        val SHOW_PROGRESS = Loading(true)
+        val HIDE_PROGRESS = Loading(false)
```

Comment:
No need for these, just you need to know when it is loading. You are already hiding the loading view when the data is successful or failed: 
![image](https://github.com/openMF/mifos-mobile/assets/49963168/9d625f36-b3fe-4ebe-ba9c-bd4244babe25)

I hope it makes sense to you!

So just add:

`object Loading: TransactionUiState()`